Tests/Unit/InstanceTests.swift

```diff
@@ -80,4 +80,162 @@ class InstanceTests: XCTestCase {
+  func testSameQueryRefInstance() throws {
+    let dc = DataConnect.dataConnect(connectorConfig: fakeConnectorConfigOne)
+
+    let queryName = "someQuery"
+    let variableOne = VariableTypeOne(email: "aa@g.com", handle: "aa@", someNumber: 12345)
+    let variableTwo = VariableTypeOne(email: "aa@g.com", handle: "aa@", someNumber: 12345)
+
+    let refOne = dc.query(
+      name: queryName,
+      variables: variableOne,
+      resultsDataType: Int.self,
+      publisher: .observableObject
+    ) as! QueryRefObservableObject<Int, VariableTypeOne>
+
+    let refTwo = dc.query(
+      name: queryName,
+      variables: variableTwo,
+      resultsDataType: Int.self,
+      publisher: .observableObject
+    ) as! QueryRefObservableObject<Int, VariableTypeOne>
+
+    let isSameInstance = refOne === refTwo
+    XCTAssertTrue(isSameInstance)
+  }
+
+  // same query name but different variable values (same variable type).
+  // should return different instances
+  func testQueryDifferentVariablesDifferentRefInstance() throws {
+    let dc = DataConnect.dataConnect(connectorConfig: fakeConnectorConfigOne)
+
+    let queryName = "someQuery"
+    let variableOne = VariableTypeOne(email: "aa1@g.com", handle: "aa@", someNumber: 12345)
+    let variableTwo = VariableTypeOne(email: "aa2@g.com", handle: "aa@", someNumber: 12345)
+
+    let refOne = dc.query(
+      name: queryName,
+      variables: variableOne,
+      resultsDataType: Int.self,
+      publisher: .observableObject
+    ) as! QueryRefObservableObject<Int, VariableTypeOne>
+
+    let refTwo = dc.query(
+      name: queryName,
+      variables: variableTwo,
+      resultsDataType: Int.self,
+      publisher: .observableObject
+    ) as! QueryRefObservableObject<Int, VariableTypeOne>
+
+    let isDifferentInstance = refOne !== refTwo
+    XCTAssertTrue(isDifferentInstance)
+  }
+
+  // different query names, same variables values
+  // should return different instances
+  func testDifferentQueryNamesSameVariables() throws {
+    let dc = DataConnect.dataConnect(connectorConfig: fakeConnectorConfigOne)
+
+    let queryNameOne = "someQueryOne"
+    let queryNameTwo = "someQueryTwo"
+    let variableOne = VariableTypeOne(email: "aa@g.com", handle: "aa@", someNumber: 12345)
+    let variableTwo = VariableTypeOne(email: "aa@g.com", handle: "aa@", someNumber: 12345)
+
+    let refOne = dc.query(
+      name: queryNameOne,
+      variables: variableOne,
+      resultsDataType: Int.self,
+      publisher: .observableObject
+    ) as! QueryRefObservableObject<Int, VariableTypeOne>
+
+    let refTwo = dc.query(
+      name: queryNameTwo,
+      variables: variableTwo,
+      resultsDataType: Int.self,
+      publisher: .observableObject
+    ) as! QueryRefObservableObject<Int, VariableTypeOne>
+
+    let isDifferentInstance = refOne !== refTwo
+    XCTAssertTrue(isDifferentInstance)
+  }
+
+  // same query name, same variables, should return same instance
```

Comment:
nit: "same query name" -> "same mutation name"

Reply:
Fixed this. Thanks!